### PR TITLE
Add dependency-injection support to LayoutRegistry

### PR DESCRIPTION
**Pull request by CamilleLetavernier**
Modify the LayoutRegistry to support LayoutRegistrations via dependency-injection, similar to how it's done for other registries

### CODE REVIEW

**Comment by CamilleLetavernier** [comment]
I'm not sure how the default layouts should be registered. Ideally, we want to be able to remove them, or to associate a different implementation with the same key (e.g. providing our own implementation for the "vbox" layout).

File: src/features/bounds/layout.ts [hunk 1]
```diff
@@ -14,27 +14,34 @@
...
 export class LayoutRegistry extends InstanceRegistry<ILayout> {
-    constructor() {
+    constructor(@multiInject(TYPES.LayoutRegistration) @optional() layouts: (LayoutRegistration)[] = []) {
         super();
-        this.register(VBoxLayouter.KIND, new VBoxLayouter());
-        this.register(HBoxLayouter.KIND, new HBoxLayouter());
-        this.register(StackLayouter.KIND, new StackLayouter());
+        layouts.forEach(layout => {
+            if (this.hasKey(layout.layoutKind)) {
+                console.log('Layout kind is already defined: ', layout.layoutKind);
```
Comment (spoenemann) [inline src/features/bounds/layout.ts:32]: Log messages should be printed via the `ILogger` service.

**Review by spoenemann — COMMENTED** [review]
Hi @CamilleLetavernier, thanks for the contribution!

You can look at the action handler registrations to see how to define the registration in a way that the default layouts can be overridden:

**Comment by spoenemann** [comment]
Yes, that makes sense! In that case it can be expressed more simply:
```typescript
bind(MoveMouseListener).toSelf().inSingletonScope();
bind(TYPES.MouseListener).toService(MoveMouseListener); 
```

**Comment by CamilleLetavernier** [comment]
Thanks for the reviews!

I've addressed the simple comments (Using logger & sorting the TYPES). I also noticed that the tests were broken, because manually instantiating a LayoutRegistry no longer registers the 3 default layouts; it should be fixed now.

I still need to add a "configureLayout" function, similar to the Action handlers, but I won't have time to do that (and test that with my use cases) properly this week; so I'll keep it for later :)

**Review by spoenemann — APPROVED** [review]
Great!

**Comment by spoenemann** [comment]
> Maybe we should apply this pattern to all classes that are multiinjected. I haven't had a detailed look yet, but I know that there at least some multiinjected classes that are not bound to itself before they are bound to the multiinjected key.

@tortmayr is there already an issue about this?

**Comment by tortmayr** [comment]
> > Maybe we should apply this pattern to all classes that are multiinjected. I haven't had a detailed look yet, but I know that there at least some multiinjected classes that are not bound to itself before they are bound to the multiinjected key.
> 
> @tortmayr is there already an issue about this?

I don't think so. I created https://github.com/eclipse/sprotty/issues/239 to track this.

**Pull request merged**